src/py42/modules/securitydata.py

```diff
@@ -169,6 +172,65 @@ def search_file_events(self, query):
         file_event_client = self._microservices_client_factory.get_file_event_client()
         return file_event_client.search(query)
 
+    def _search_by_hash(self, hash, type):
+        query = FileEventQuery.all(type.eq(hash))
+        response = self.search_file_events(query)
+        return response["fileEvents"]
+
+    def _find_file_versions(self, md5_hash, sha256_hash):
+        file_event_client = self._microservices_client_factory.get_file_event_client()
+        pds_client = self._microservices_client_factory.get_preservation_data_service_client()
+        for device_id, paths in file_event_client.get_file_location_detail_by_sha256(sha256_hash):
+            yield pds_client.find_file_versions(md5_hash, sha256_hash, device_id, paths)
+
+    def _stream_file(self, file_generator, file):
+        for response in file_generator:
+            if response.status_code != 200:
+                continue
+            session = self._microservices_client_factory.get_storage_node_session(
+                response[u"storageNodeURL"]
+            )
+            storage_node_client = StorageNodeClient(
+                session,
+                response[u"archiveGuid"],
+                response[u"fileId"],
+                response[u"versionTimestamp"],
+            )
+            token = storage_node_client.get_download_token()
+            return storage_node_client.get_file(response[u"storageNodeURL"], token, file)
+        raise FileNotFoundError("No file available for download.")
+
+    def stream_file_by_sha256(self, checksum):
```

Comment:
If writing data to file is fine, do we want user to pass additional parameter, optionally, "path with filename" where data needs to be downloaded? 

Right now it writes to a file with the given checksum.

Reply:
Per above, simply return the stream and do not write a file.

Reply:
Done.